pkg/http/core/artifacts_test.go

```diff
@@ -55,6 +55,20 @@ var _ = Describe("Artifacts", func() {
 			doRequest()
 		})
 
+		When("getting the helm client returns an error", func() {
+			BeforeEach(func() {
+				fakeArtifactCredentialsController.HelmClientForAccountNameReturns(nil, errors.New("error getting helm client"))
+			})
+
+			It("returns an error", func() {
+				Expect(res.StatusCode).To(Equal(http.StatusBadRequest))
+				ce := getClouddriverError()
```

Comment:
Can you move this line to the BeforeEach block?

Reply:
That func is reading from the http response, so I need to call that func after the http req has been made.

Reply:
That's fine, Would JustBeforeEach work in that case?

Reply:
Technically yes. The first jbe makes the request so internal jbe could read from the response, however I think that declaring a new variable scoped to the whole when block and adding a jbe to grab this is less readable and results in more lines. It's also more prone to user errors in the future.

If we decide to go this route of declaring and grabbing the clouddriver error in a jbe we should do this project-wide under another branch - but I think we should leave it as is for now.

Reply:
Agreed